script.js

```diff
@@ -0,0 +1,60 @@
+const arrRates = []
+
+document.querySelector('.formRate--js').addEventListener('submit', getRate);
+document.querySelector('.formConvert--js').addEventListener('submit', getConvert)
+
+function getRate(event) {
+   event.preventDefault();
+   const currencyFrom = document.getElementById('currency_from').value;
+   const currencyTo = document.getElementById('currency_to').value;
+   const date = document.getElementById('currency_date').value;
+   const rate = +(document.getElementById('currency_rate').value);
+   
+   let existingRate = arrRates.find(rateObj => rateObj.base === currencyFrom);
+
+   if (existingRate) {
+      existingRate.rates[currencyTo] = rate; //update existing currency conversion with a new rate
+   } else {
+      arrRates.push({
+         timestamp: Date.now(),
+         base: currencyFrom,
+         date: date,
+         rates: { [currencyTo]: rate }
+      });
+   }
+   console.log(arrRates) 
+}
+
+function getConvert(event) {
+   event.preventDefault();
+   const amount = +(document.getElementById('amount').value);
+   const convertCurrencyFrom = document.getElementById('convert-currency_from').value;
+   const convertCurrencyTo = document.getElementById('convert-currency_to').value;
+   let convertedAmount = null;
+
+   function getKeyObj(obj, currencyNameTo) {
```

Comment:
Great! You managed to process the result and find the conversion of your saved objects, very well! 🤩

🤓 Just to add an alternative way to approach this issue:

For example,

getKeyObj and getValueObj

These functions seem designed to search for a specific key in an object and return its corresponding value. However, in this context, these functions are not necessary and can be eliminated, as the goal can be achieved directly by accessing the object's properties.

For instance, by using a loop:

```
 for (let i = 0; i < arrRates.length; i++) {
    const rateItem = arrRates[i];
    if (rateItem.base === convertCurrencyFrom && rateItem.rates[convertCurrencyTo]) {
        const rate = rateItem.rates[convertCurrencyTo];
        convertedAmount = amount * rate;
        if (convertedAmount !== null) {
            document.getElementById("result").innerHTML = convertedAmount;
        } else return        
    }
}
```